src/components/PageHeader.astro

```diff
@@ -0,0 +1,59 @@
+---
+import { PAGES_CONFIG } from '@/constants/pages';
+import { Typewriter } from '@components/Name.tsx';
+import { NavBar } from '@components/NavBar.tsx';
+import { getPagesLinks } from '@/helpers/pages';
+import type { PageKey } from '@/types/pages';
+
+interface PageHeaderProps {
+  pageKey: PageKey;
+}
+
+const { pageKey } = Astro.props as PageHeaderProps;
+---
+
+<div
+  class="flex flex-col flex-auto content-center justify-start items-center pointer-none"
+>
+  <Typewriter
+    client:load
+    text={[PAGES_CONFIG[pageKey].text]}
+    delay={100}
+    isSmall
+  />
+  <hr class="separator separator--line" />
+  <NavBar client:only="react" links={getPagesLinks({ excludeKey: pageKey })} />
+</div>
```

Comment:
I started using this across all pages (except `FrontPage`) so I extract it into a component